client/src/containers/SubmitCodeSmells.js

```diff
@@ -37,11 +34,21 @@ class SubmitCodeSmells extends Component {
 
     handleSubmit() {
 		const { dispatch, code, codeid, userid, selectedLines } = this.props;
-        if (selectedLines.length !== 0) {
-            dispatch(submitCode(userid, code, selectedLines));
-        } else {
-            console.log("You didn't input any code smells!");
-        }
+		if (!this.state.hasSubmitted) {
+			if (selectedLines.length !== 0) {
+				dispatch(submitCode(userid, code, selectedLines));
+				this.setState({
```

Comment:
it's probably not guaranteed that the submit will work but you open the modal any way. Instead maybe wait for component with receive props for a successful has submitted then open the modal. But it's a bit more complicated. Up to your digression if you want to do it that way.